### ISSUE TITLE
Functionality restricted by lack of communication between 'Manage Current Task' and 'Visualisation RM WtdAv'

### ISSUE THREAD

**Issue by sbodorkos** [CIRDLES/Squid#313]
I see this as a relatively minor issue related to how the interface works, but I am raising it now because it is impeding some of the 'housekeeping' I need to do as part of the arithmetical comparisons, and I think the problem affects all Perms except Perm1.

The illustration is with a Perm2 Task. The screenshot below shows the relevant 'Manage Current Task' screen, with my Parameters specified the way I want (index isotope = 207; Auto-reject spots = NO, and 'Refresh' pressed):

![image](https://user-images.githubusercontent.com/17473203/56323367-f8ed0f00-61ae-11e9-958e-b88bd4440e6a.png)

When I go directly to the Visualisations menu, and select Reference Materials... Weighted Means, this is what I get (every time):

![image](https://user-images.githubusercontent.com/17473203/56323591-b0822100-61af-11e9-9d22-d7eb58a7a899.png)

Note that in this screen, the correction **always** defaults to 4-corr, and auto-reject spots **always** defaults to YES, even though both of these contravene my stated preferences in 'Manage Current Task'.

This in itself is no big deal. Obviously I can switch the Correction radio-button directly in this screen. The 'Auto-reject spots' behaviour is a bit more tedious; I need to return to 'Manage Current Task', set 'Auto-reject spots' to YES, and then **immediately set it back to NO**, in order to get the instruction to "take". But that does work; I do get what I want in the end, which is this:

![image](https://user-images.githubusercontent.com/17473203/56324164-7ca7fb00-61b1-11e9-89d9-25392a504323.png)

So, two relatively simple things to significantly improve useability here:

1. Rigorous, instantaneous synchronisation of the 'Preferred index isotope' radio-button in 'Manage Current Task', and the 'Correction' radio-button in 'Reference Material Weighted Mean' i.e. it should be bidirectional; changing the setting in one window ought to change the same setting in the other.

2. Place a "copy" of the 'Auto-reject spots' check-box in the Reference Material Weighted Mean' screen, and synchronise that in the same way. (In fact, it could be argued that this check-box should reside **solely** in the Reference Material Weighted Mean window, as it isn't really a Task attribute at all, as per issue #291 ).

_And if you were after some WOW factor, the other radio-buttons corresponding to 'Manage Current Task' Parameters could be copied (or transplanted) to Reference Material Weighted Mean and synchronised in the same way. It would bring those controls to "life" if you could change the settings whilst directly observing their effect on the calibration, via Reference Material Weighted Average._ 

_Not just cool, but also a really good way of showcasing the power of what you have already done, in terms of enabling 'on-the-fly' recalculations. Analogous alterations in SQUID 2.50 require reprocessing of the Prawn XML file "from the top", and everyone knows this, so it would be a good way of reminding the broader user-group that Squid3 contains functional enhancements as well as arithmetical replication._

**BUT I DIGRESS** - the functionality restriction mentioned in the issue title relates to analogous behaviour in the 208Pb/232Th calibration constant (i.e. the other Flavour).

If I want to view Reference Material Weighted Mean for Correction = 7-corr, Flavour = 208Pb/232Th, and 'Auot-reject spots' = NO (in preparation for creating a Report Table using those parameters), I can't actually do that.

Switching Flavour and Correction via radio-button still gives a diagram in which 'Auto-reject spots' = YES. It looks like this:

![image](https://user-images.githubusercontent.com/17473203/56339667-87cf4b00-61f2-11e9-86a5-04fcecd9c3de.png)

Problem is, now I need to go back to Manage Current Task, and switch 'Auto-reject Spots' = YES and then NO again, then return to Reference Material Weighted Mean. When I return, I am back to the 206/238 Flavour (and 4-corr Correction) by default, and although that combination (temporarily) honours my choice of 'Auto-reject', switching Flavour to 208Pb/232Th disregards that selection all over again, and my 208Pb/232Th Weighted Mean plot reverts once again to 'Auto-reject' = YES. 

This means that the option of applying 'Auto-reject' = NO to 208Pb/232Th calibrations in Perms 2, 3 and 4 is currently not available to Squid3 users... and I had intended to use that option for my arithmetical comparison... so a short-term workaround would be good.

I think this highlights a couple of points for consideration:

1. This is a fundamental screen in the data-reduction and reporting process. In fact, you can't generate Report Tables with any visibility/understanding of what you're doing, without visiting this screen. It could be argued that the Report generation process should be "nested" here. The current menu structure gives the impression that Reports can meaningfully be generated prior to/independent of the Visualisations, which isn't true.

2. As such, it's really more than just a Visualisation, because the associated Weighted Mean data dictate the content of the Unknowns Report-Tables. I think it is necessary to upgrade the "Flavour" radio-button to a formal two-tab structure, where the left-hand tab always portrays and manipulates the 206Pb/238U calibration where applicable (it would be deactivated for Perm3), and the right-hand tab always portrays and manipulates the 208Pb/232Th calibration (deactivated for Perm1). Each tab would have its own 'Auto-reject spots' control, but any other radio-buttons (SBM normalisation, LinReg vs SpotAvg etc.) would need to apply equally to both. The details can be negotiated, but it will be important to cement in the minds of users that in the Perm2 and Perm4 cases especially the two Wtd Means both materially affect the contents of the Report Tables, and perusing one without checking the other is unlikely to lead them to a good place...

None of this is an immediate priority. But it's important to properly recognise the convergence-points in the processing sequence, because these are what we will use to restructure the interface to actually lead users through the process of properly reducing their data. The Reference Materials Weighted Mean screen is just as important/compulsory as the Manage Spots and Reference Materials screen, and should be used in the same way, as a site for "nesting" related functions (Report-Tables in this case).

**Comment by bowring** [comment]
Pull request #314 addresses the first half of this discussion through items 1 and 2:
Fixes a bug that caused the miscommunication of the allows rejection choice and introduces two features requested.  1) the selected index isotope works in both directions between the weighted mean and the task, but this raises questions about the architecture in general - for example, there is NO 7-corr Concordia for the ref material, so the two-way mechanism is not really robust; 2) the auto-reject mechanism is now two-way between weighted means and task;

**Comment by sbodorkos** [comment]
Yes, you are correct about current lack of robustness. Pursued to the bitter end, the RM Concordia screen should go blank when "7-corr" is selected... but there are alternatives to that (e.g. plotting data uncorrected for common Pb, which is a common mechanism for assessing the appropriateness of the 207-correction anyway), but that's a job for another day...

The current situation has arisen though an "evolution" in our understanding of the role of the index isotope. To begin with, we instituted the index-isotope radio-button primarily as a visualisation control, but what we have learned in the 12-18 months since is that the index isotope exerts minor but widespread control on many aspects of the data-reduction arithmetic. In truth, it is a "directive" of sorts, albeit not Task-specific. 

One way forward would be to have dissociated index-isotope controls; one that dictates data-reduction, and another used solely for visualisations. But then it would be necessary to rigorously specify how the two interact. I think it is probably preferable to have a single control that helps highlight the consequences of index-isotope choice. Users should already be aware that choosing the 207-correction precludes visualisation of their common-Pb corrected RM data on any kind of Concordia plot - but if they are not aware, we can certainly improve their understanding!
